Modules/_sqlite/connection.c

```diff
@@ -79,40 +79,34 @@ new_statement_cache(pysqlite_Connection *self, int maxsize)
     return res;
 }
 
+/*[clinic input]
+_sqlite3.Connection.__init__ as pysqlite_connection_init
+
+    database as database_obj: object(converter='PyUnicode_FSConverter')
+    timeout: double = 5.0
+    detect_types: int = 0
+    isolation_level: object = NULL
+    check_same_thread: bool(accept={int}) = True
+    factory: object(c_default='(PyObject*)clinic_state()->ConnectionType') = ConnectionType
```

Comment:
Hummmm, i don't fully get how this is preserving the state as before. Seems that the previous code didn't have a default for factory and used NULL as the c default. Could you briefly explain how this default is preserving previous behaviour?

Reply:
Sure :) This is the current code: https://github.com/python/cpython/blob/1d10bf0bb9409a406c56b0de8870df998637fd0f/Modules/_sqlite/module.c#L92-L95

In practice, the `factory` default is `ConnectionType`, not `NULL`, so I chose to let AC take care of this instead of writing it out explicitly in the function body. This also aligns well with the documentation, which says that `factory` defaults to the `sqlite3.Connection` type.

Reply:
Oh, I see. Is other code using this branch in `module.c:92`? I winder if we can enforce that a factory is always provided after this PR

Reply:
> Is other code using this branch in module.c:92?

Not in the stdlib (except for some tests in the test suite), but I guess it's used in the wild. Personally, I would not have added this feature in the first place.

> I winder if we can enforce that a factory is always provided after this PR

That's kind of what we're doing no with the default AC argument, right?